wallet-new/integration/Functions.hs

```diff
@@ -677,25 +700,25 @@ chooseActions n probs = liftIO . generate $ do
 respToRes
     :: forall m a. (MonadThrow m)
     => m (Either ClientError (WalletResponse a))
-    -> m a
+    -> PropertyM m a
 respToRes resp = do
-    result <- resp
-    either throwM (pure . wrData) result
+    result <- lift resp
+    run $ either Universum.throwM (pure . wrData) result
 
 
 -- | Pick a random element using @IO@.
-pickRandomElement :: (MonadIO m, Alternative m) => [a] -> m a
-pickRandomElement [] = empty
-pickRandomElement xs = liftIO . generate . elements $ xs
+pickRandomElement :: (Show a, MonadIO m) => [a] -> PropertyM m a
+pickRandomElement [] = liftIO empty
```

Comment:
This isn't really what we want. This will throw an `IOError` in the `IO` monad if we have an empty list and won't give any error reporting information at all!

The old implementation would discard the test. However, `pre :: Bool -> PropertyM m ()` forces `()`. I would suggest looking at the source code for `pre` and figure out how to write a `reject :: PropertyM m a` that discards the current test case if it is called.